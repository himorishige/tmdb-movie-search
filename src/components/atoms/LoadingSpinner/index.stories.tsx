import { Meta, Story } from '@storybook/react';
import LoadingSpinner from '.';

export default {
  title: 'TMDB Search/Atoms/LoadingSpinner',
  component: LoadingSpinner,
} as Meta;

export const Default: Story = (args) => <LoadingSpinner {...args} />;
