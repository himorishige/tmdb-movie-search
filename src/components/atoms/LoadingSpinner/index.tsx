import { memo } from 'react';
import { Box, Flex, Spinner } from '@chakra-ui/react';

const LoadingSpinner: React.VFC = memo(() => {
  return (
    <Flex justifyContent="center" alignItems="center">
      <Box>
        <Spinner thickness="4px" speed="0.65s" emptyColor="gray.200" color="blue.500" size="xl" />
      </Box>
    </Flex>
  );
});

export default LoadingSpinner;
