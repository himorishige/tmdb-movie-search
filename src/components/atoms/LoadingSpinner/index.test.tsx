import renderer from 'react-test-renderer';
import LoadingSpinner from '.';

describe('Atoms/LoadingSpinner', () => {
  test('正しくレンダリングされている', () => {
    const spinner = renderer.create(<LoadingSpinner />).toJSON();
    expect(spinner).toMatchSnapshot();
  });
});
