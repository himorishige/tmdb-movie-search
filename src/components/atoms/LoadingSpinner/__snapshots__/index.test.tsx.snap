// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/LoadingSpinner 正しくレンダリングされている 1`] = `
<div
  className="css-gmuwbf"
>
  <div
    className="css-0"
  >
    <div
      className="chakra-spinner css-u4uvb9"
    >
      <span
        className="css-f8n5zr"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
